cilium/CFP-42466-change-cilium-pod-ip-pool-crd.md

```diff
@@ -0,0 +1,102 @@
+# CFP-42466: change-multi-ip-pool-crd
+
+**SIG: SIG-NAME** ([View all current SIGs](https://docs.cilium.io/en/stable/community/community/#all-sigs))
+
+**Begin Design Discussion:** 2025-11-04
+
+**Cilium Release:** 1.19
+
+**Authors:** kyounghoonJang <matkimchi_@naver.com>
+
+**Status:** Draft
+
+**Reference** this CFP was translated from Korean to English using AI.
+
+## Summary
+
+The current CiliumPodIPPool CRD represents CIDRs as simple string lists, preventing granular control over allocation.
+This proposal introduces a schema enhancement that supports marking specific CIDRs as non-allocatable — addressing issues like orphaned IPs and enabling gradual, safe IP pool decommissioning.
+
+## Motivation
+
+When using Multi-Pool IPAM, operators sometimes need to decommission or replace existing CIDRs.
+The only current way to stop IP allocation from a CIDR is to remove it from the pool entirely, which can cause existing pod IPs to become orphaned and trigger warnings in the operator.
+
+To support a two-step migration workflow, we propose enabling CIDRs to be marked as “non-allocatable”:
+
+1. Add a new CIDR and mark the old CIDR as non-allocatable.
+
+2. Drain and migrate workloads to nodes using the new CIDR.
+
+3. Once the old CIDR is no longer in use, remove it from the pool.
+
+This allows for controlled and predictable transitions between IP pools without service disruption.
+
+## Goals
+Introduce a new `unallocatable` flag in the `CiliumPodIPPool` CRD to mark specific CIDRs as non-allocatable.
+
+## Proposal
+
+### Overview
+
+Current structure:
+```yaml
+apiVersion: cilium.io/v2alpha1
+kind: CiliumPodIPPool
+metadata:
+  name: green-pool
+spec:
+  ipv4:
+    cidrs:
+      - 10.20.0.0/16
+      - 10.30.0.0/16
+    maskSize: 24
+```
+Proposed structure:
+```yaml
+apiVersion: cilium.io/v2alpha1
```

Comment:
I think this should be `cilium.io/v2` - I don't think we can change the type of the `cidrs` list type otherwise, because it would break all users using `v2alpha1`.